powerline_shell/segments/ruby_version.py

```diff
@@ -2,21 +2,31 @@
 import subprocess
 from ..utils import BasicSegment
 
+FANCY_RUBY = u'\ue791'
+
+def _get_ruby_version():
+    try:
+        p1 = subprocess.Popen(['ruby', '-v'], stdout=subprocess.PIPE)
+        p2 = subprocess.Popen(['sed', "s/ (.*//"], stdin=p1.stdout, stdout=subprocess.PIPE)
+        return p2.communicate()[0].decode('utf-8').rstrip()
+    except OSError:
+        raise
 
 class Segment(BasicSegment):
     def add_to_powerline(self):
         powerline = self.powerline
 
         try:
-            p1 = subprocess.Popen(['ruby', '-v'], stdout=subprocess.PIPE)
-            p2 = subprocess.Popen(['sed', "s/ (.*//"], stdin=p1.stdout, stdout=subprocess.PIPE)
-            ruby_and_gemset = p2.communicate()[0].decode('utf-8').rstrip()
+            ruby_and_gemset = _get_ruby_version()
+        except OSError:
+            return
 
-            gem_set = os.environ.get('GEM_HOME', '@').split('@')
+        if powerline.segment_conf("ruby_version", "mode") == "fancy":
+            ruby_and_gemset = ruby_and_gemset.replace('ruby ', FANCY_RUBY)
 
-            if len(gem_set) > 1:
-                ruby_and_gemset += "@{}".format(gem_set.pop())
+        gem_set = os.environ.get('GEM_HOME', '').split('@')[1:]
```

Comment:
Note that this method of getting the name of the gemset covers all three test cases added to the ruby_version segment test below.